Documentation/Books/Manual/DataModeling/OperationalFactors.md

```diff
@@ -0,0 +1,141 @@
+
+There are ways to ensure atomicity and consistency when perfoming updates in your application.
+ArangoDB allows you to specify the revision ID of the existing document you would like to update.
+This way you can ensure that if your application has read a document with a certain revision, the
+modifications to it are only allowed to pass if and only if the document was not changed by someone else.
+
+You can specify the revision via the `_rev` field inside the document or via the `If-Match: <revision>`
+header in the documents REST API. In the _arangosh_ you can peform such an operation like this:
+
+```js
+db.basketCollection.update({"_key": "123", "_rev": "_Xv0TA0O--_"}, data)
+// or replace
+db.basketCollection.replace({"_key": "123", "_rev": "_Xv0TA0O--_"}, data)
+```
+
+Indexes
+-------
+
+Indexes can improve the performance of AQL queries drastically. Queries that frequently filter on one more fields
+can be made faster by creating an index via the _ensureIndex_ command. There is already a permanent primary
+index in every collection on the *_key* and *_id* fields as well as the edge index on *_from* and *_to* (for edge collections).
+
+Should you decide to create an index you should consider a few things:
+
+- Indexes are a trade-off between storage space and query speed.
+- Each new index will increase the amount of RAM and (for the rocksdb storage) the amount of disk space needed.
+- Indexes with [indexed array values](../Indexing/IndexBasics.md#indexing-array-values) need an extra index entry per array entry
+- Consider the extra space requirements of indexes when planning server capacities.
+- Adding indexes increases the write-amplification i.e. it negatively affects the write performance
+- Each index needs to add at least one index entry per document. You can use _sparse indexes_ to avoid adding index entries for unused fields
+- Collections that are more frequently read benefit the most from added indexes, provided you actually use the indexes
+- Collections with a high rate of inserts or updates compared to reads may hurt overall performance.
+
+Generally it is best to design your indexes with your queries in mind. Use the [query profiler](../../AQL/ExecutionAndPerformance/QueryProfiler.html) to understand the bottlenecks in your queries. For more information on indexes see [Index Basics](../Indexing/IndexBasics.md)
+
+<!-- TODO maybe expand a bit on capacity planning ? -->
+
+Number of Databases and Collections
+-----------------------------------
+
+Sometimes you can consider to split up data over multiple collections. For example one could create a
+new set of collections for each new customer, instead of having a customer field on each documents. Having a
+large number of collections has no performance penalty and results in good performance.
+
+Grouping documents into collections by type (i.e. a session collection 'sessions_dev', 'sessions_prod') allows
+you to avoid an index on the type field.
```

Comment:
Could also suggest to split collections based on the attributes used, which may indicate that different types of entities are stored in one collection. Might allow to turn sparse indexes to non-sparse if the indexed field is used for splitting.